rust/otap-dataflow/crates/engine/src/control.rs

```diff
@@ -9,12 +9,91 @@ use crate::error::{Error, TypedError};
 use crate::message::Sender;
 use crate::node::{NodeId, NodeType};
 use crate::shared::message::{SharedReceiver, SharedSender};
+use bytemuck::Pod;
 use otap_df_channel::error::SendError;
 use otap_df_telemetry::reporter::MetricsReporter;
+use smallvec::SmallVec;
 use std::collections::HashMap;
 use std::marker::PhantomData;
 use std::time::Duration;
 
+/// A 8-byte context value. Supports conversion to and from plain data
+/// using bytemuck.
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug)]
+pub struct Context8u8([u8; 8]);
+
+impl<T: Pod> From<T> for Context8u8 {
+    /// From T to Context8u8
+    fn from(v: T) -> Self {
+        Self(bytemuck::cast(v))
+    }
+}
```

Comment:
This can panic if `T` isn't 8 bytes long https://docs.rs/bytemuck/1.24.0/bytemuck/fn.cast.html

> This is like [try_cast](https://docs.rs/bytemuck/1.24.0/bytemuck/fn.try_cast.html), but will panic on a size mismatch.

For example:
```rs
        let x = 5u64;
        let y = Context8u8::from(x); // <-- this is OK
        println!("{:?}", y);

        let x2 = 5u16;
        let y2 = Context8u8::from(x2); // <-- this will panic
        println!("{:?}", y2);
```

Maybe we could add a compile time assertion that the length of `T` is correct?
```suggestion
impl<T: Pod> From<T> for Context8u8 {
    /// From T to Context8u8
    fn from(v: T) -> Self {
        const {
            assert!(size_of::<T>() == 8);
        }
        Self(bytemuck::cast(v))
    }
}
```